src/chunk.c

```diff
@@ -910,6 +937,21 @@ chunk_scan_ctx_get_chunk(ChunkScanCtx *ctx)
 	return ctx->data;
 }
 
+/* Gets all the chunks from the given scan context. Used for time constraint
+ * elimination and that is why we don't care about constraints in
+ * other dimensions.
+ * Returns NIL if no chunks were found. Returns the list of Chunk* objects
+ * otherwise.
+ */
+static List *
+chunk_scan_ctx_get_all_chunks(ChunkScanCtx *ctx)
+{
+	ctx->data = chunks_alloc(CHUNKS_DEFAULT_CAPACITY);
```

Comment:
Here you can just allocate with the number of entries in the hash table within the scan context. I am not sure you even need the expandable `Chunks` container. I think you might be able to get away with a fixed size `Chunk *` array without all the boilerplate code for the `Chunks` type. See also my comment above about this.